draft-ietf-quic-version-negotiation.md

```diff
@@ -68,6 +68,73 @@ document are to be interpreted as described in BCP 14 {{!RFC2119}} {{!RFC8174}}
 when, and only when, they appear in all capitals, as shown here.
 
 
+# Server Deployments of QUIC {#server-fleet}
+
+While this document mainly discusses a single QUIC server, it is common for
+deployments of QUIC servers to include a fleet of multiple server instances. We
+therefore define the following terms:
+
+Accepted Versions:
+
+: This is the set of versions supported by a given server instance. More
+specifically, these are the versions that a given server instance will use if a
+client sends a first flight using them.
+
+Negotiated Versions:
+
+: This is the set of versions that a given server instance will send in a
+Version Negotiation packet if it receives a first flight from an unknown
+version. This set will most often be equal to the Accepted Versions set, except
+during short transitions while versions are added or removed (see below).
+
+Fully-Deployed Versions:
+
+: This is the set of QUIC versions that is supported and negotiated by every
+single QUIC server instance in this deployment. If a deployment only contains a
+single server instance, then this set is equal to the Negotiated Versions set,
+except during short transitions while versions are added or removed (see below).
+
+If a deployment contains multiple server instances, it is possible for software
+updates to not happen at the exact same time on all server instances. Because
+of this, a client might receive a Version Negotiation packet from a server
+instance that has already been updated and the client's resulting connection
+attempt might reach a different server instance which hasn't been updated yet.
+
+However, even when there is only a single server instance, it is still possible
+to receive a stale Version Negotiation packet if the server performs its
+software update while the Version Negotiation packet is in flight.
+
+This could cause the version downgrade prevention mechanism described in
+{{downgrade}} to falsely detect a downgrade attack. To avoid that, server
+operators SHOULD perform a three-step process when they wish to add or remove
+support for a version:
+
+* When adding support for a new version, the first step is to progressively add
+  support for the new version to all server instances. This step updates the
+  Accepted Versions but not the Negotiated Versions nor the Fully-Deployed
+  Versions. Once all server instances have been updated, operators wait for at
+  least one minute to allow any in-flight Version Negotiation packets to
+  arrive. Then, the second step is to progressively add the new version to
+  Negotiated Versions on all server instances. Once complete, operators wait
+  for at least another minute. Finally, the third step is to progressively add
+  the new version to Fully-Deployed Versions on all server instances.
```

Comment:
```suggestion
support for a version.

When adding support for a new version:

- The first step is to progressively add
  support for the new version to all server instances. This step updates the
  Accepted Versions but not the Negotiated Versions nor the Fully-Deployed
  Versions. Once all server instances have been updated, operators wait for at
  least one minute to allow any in-flight Version Negotiation packets to
  arrive. 
- Next, progressively add the new version to
  Negotiated Versions on all server instances. Once complete, operators wait
  for at least another minute.
- Finally, progressively add
  the new version to Fully-Deployed Versions on all server instances.
```
Modulo rewrapping, I think moving the bullets down a level is clearer.

Reply:
Fixed